frontend/src/my-books/MyBooks.tsx

```diff
@@ -15,25 +15,80 @@ You should have received a copy of the GNU General Public License along with thi
 If not, see <https://www.gnu.org/licenses/>.
 */
 
-import React, { Component } from 'react'
-import { ShelfCarousel } from '../shared/book-display/ShelfCarousel'
-import { Layout } from "../shared/components/Layout";
-import './MyBooks.css'
+import React, { Component, ReactElement } from "react";
+import { NavBar } from "../shared/navigation/NavBar";
+import { ShelfCarousel } from "../shared/book-display/ShelfCarousel";
+import "./MyBooks.css";
+import Button from "@material-ui/core/Button";
+import ShelfModal from "./ShelfModal";
+import Hidden from "@material-ui/core/Hidden";
+import MenuIcon from "@material-ui/icons/Menu";
 
 interface IState {
+    showShelfModal: boolean;
 }
 
 class MyBooks extends Component<Record<string, unknown>, IState> {
-    render() {
+    constructor(props: Record<string, unknown>) {
+        super(props);
+        this.state = {
+            showShelfModal: false,
+        };
+        this.onAddShelf = this.onAddShelf.bind(this);
+        this.onAddShelfModalClose = this.onAddShelfModalClose.bind(this);
+    }
+
+    onAddShelf(): void {
+        this.setState({
+            showShelfModal: true,
+        });
+    }
+
+    onAddShelfModalClose(): void {
+        this.setState({
+            showShelfModal: false,
+        });
+    }
+    render(): ReactElement {
         return (
-            <Layout title="My books">
-                <ShelfCarousel title="Reading" />
-                <ShelfCarousel title="To Read" />
-                <ShelfCarousel title="Read" />
-                <ShelfCarousel title="Did not finish" />
-            </Layout>
-        )
+            <div>
+                <NavBar />
+                <Hidden smDown implementation="css">
```

Comment:
Remove Hidden tag here. The buttons should show on mobile. We don't yet have responsive designs so for now, we can just have them showing.